qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -649,17 +649,26 @@ def on_current_changed(self, idx):
             log.webview.debug("on_current_changed got called with invalid "
                               "index {}".format(idx))
             return
+        if self._now_focused and config.val.tabs.mode_on_change == 'restore':
+            current_mode = modeman.instance(self._win_id).mode
+            # only save insert & passthrough mode. otherwise default to normal
```

Comment:
Not sure if the comment is needed - it says pretty much the same the code says, and the code is pretty clear IMHO.